### PR TITLE
Chapter 4 Add a full_title helper

### PR DESCRIPTION
**Pull request by throw1280**
image

### CODE REVIEW

**Comment by throw1280** [comment]
ready

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (thanhhungchu95) [inline app/helpers/application_helper.rb]: i18n

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title><%= yield :title %><%= t :title_app %></title>
+    <title><%= full_title  yield :title %></title>
```
Comment (thanhhungchu95) [inline app/views/layouts/application.html.erb]:
```suggestion
    <title><%= full_title yield :title %></title>
```

File: app/views/static_pages/about.html.erb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <% provide :title, t("about") %>
 <h1><%= t :about %></h1>
 <p>
-  <%= link_to "https://www.railstutorial.org/" do %> <em><%= t :rort %></em><%= t :isa %><% end %>
+  <%= link_to "https://www.railstutorial.org/" do %> <em><%= t :rort %></em><%= t :isa %> <% end %>
```
Comment (thanhhungchu95) [inline app/views/static_pages/about.html.erb]: chia thành 3 dòng

**Comment by throw1280** [comment]
ready

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (throw1280) [reply app/helpers/application_helper.rb]: def full_title page_title = ""
    base_title = t("title_app")
    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
  end

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (throw1280) [reply app/views/layouts/application.html.erb]: <head>
    <title><%= full_title yield :title %></title>
    <%= csrf_meta_tags %>
    <%= csp_meta_tag %>
    <%= stylesheet_link_tag "application", media: "all", "data-turbolinks-track": "reload" %>
    <%= javascript_include_tag "application", "data-turbolinks-track": "reload" %>
  </head>

File: app/views/static_pages/about.html.erb (hunk 3 above)
Comment (throw1280) [reply app/views/static_pages/about.html.erb]: <p>
  <%= link_to "https://www.railstutorial.org/" do %>
    <em><%= t :rort %></em><%= t :isa %>
  <% end %>

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (throw1280) [reply app/views/layouts/application.html.erb]: <title><%= full_title yield :title %></title>

File: app/views/static_pages/about.html.erb (hunk 3 above)
Comment (throw1280) [reply app/views/static_pages/about.html.erb]: <%= link_to "https://www.railstutorial.org/" do %>
    <em><%= t :rort %></em><%= t :isa %>
  <% end %>

File: config/environments/test.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Rails.application.configure do
+  Rails.application.configure do
```
Comment (thanhhungchu95) [inline config/environments/test.rb]: xóa 2 space ở đầu dòng nak

File: config/locales/en.yml [hunk 5]
```diff
@@ -55,4 +55,3 @@ en:
   rort: "Ruby on Rails Tutorial"
   ror: "Ruby on Rails"
```
Comment (thanhhungchu95) [inline config/locales/en.yml]: xóa dòng thừa

File: app/views/static_pages/help.html.erb [hunk 6]
```diff
@@ -4,5 +4,7 @@
   <%= t :get %>
   <%= link_to t("hp"),"https://www.railstutorial.org/book" %>
   <%= t :tot %>
-  <%= link_to "https://www.railstutorial.org/book" do %> <em><%= t :rort %></em><%=  t :book %><% end %>
+  <%= link_to "https://www.railstutorial.org/book" do %>
+   <em><%= t :rort %></em><%=  t :book %>
```
Comment (thanhhungchu95) [inline app/views/static_pages/help.html.erb]: indent 2 space

File: config/locales/en.yml (hunk 5 above)
Comment (throw1280) [reply config/locales/en.yml]: image

File: app/views/static_pages/help.html.erb (hunk 6 above)
Comment (throw1280) [reply app/views/static_pages/help.html.erb]: image

**Review by thanhhungchu95 — COMMENTED** [review]
(merged) đi em

**Pull request merged**